cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperEraser.java

```diff
@@ -0,0 +1,198 @@
+import io.strimzi.operator.cluster.ClusterOperatorConfig;
+import io.strimzi.operator.cluster.model.ZookeeperCluster;
+import io.strimzi.operator.cluster.operator.resource.ResourceOperatorSupplier;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.operator.resource.ConfigMapOperator;
+import io.strimzi.operator.common.operator.resource.NetworkPolicyOperator;
+import io.strimzi.operator.common.operator.resource.PodDisruptionBudgetOperator;
+import io.strimzi.operator.common.operator.resource.PvcOperator;
+import io.strimzi.operator.common.operator.resource.SecretOperator;
+import io.strimzi.operator.common.operator.resource.ServiceAccountOperator;
+import io.strimzi.operator.common.operator.resource.ServiceOperator;
+import io.strimzi.operator.common.operator.resource.StorageClassOperator;
+import io.strimzi.operator.common.operator.resource.StrimziPodSetOperator;
+import io.vertx.core.Future;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Class used for deleting the ZooKeeper ensemble when the ZooKeeper to KRaft migration is completed
+ * and the Kafka cluster is full KRaft, not using ZooKeeper anymore for storing metadata
+ */
+public class ZooKeeperEraser {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(ZooKeeperEraser.class.getName());
+
+    private final Reconciliation reconciliation;
+    private final ZookeeperCluster zk;
+    private final boolean isNetworkPolicyGeneration;
+    private final StrimziPodSetOperator strimziPodSetOperator;
+    private final SecretOperator secretOperator;
+    private final ServiceAccountOperator serviceAccountOperator;
+    private final ServiceOperator serviceOperator;
+    private final PvcOperator pvcOperator;
+    private final StorageClassOperator storageClassOperator;
+    private final ConfigMapOperator configMapOperator;
+    private final NetworkPolicyOperator networkPolicyOperator;
+    private final PodDisruptionBudgetOperator podDisruptionBudgetOperator;
+
+    /**
+     * Constructs the ZooKeeper eraser
+     *
+     * @param reconciliation            Reconciliation marker
+     * @param config                    Cluster Operator Configuration
+     * @param supplier                  Supplier with Kubernetes Resource Operators
+     * @param kafkaAssembly             The Kafka custom resource
+     */
+    public ZooKeeperEraser(
+            Reconciliation reconciliation,
+            ClusterOperatorConfig config,
+            ResourceOperatorSupplier supplier,
+            Kafka kafkaAssembly
+    ) {
+        this.reconciliation = reconciliation;
+        // NOTE: we are going to use the eraser for deleting ZooKeeper so we are not interested in providing old storage and replicas parameters
+        this.zk = ZookeeperCluster.fromCrd(reconciliation, kafkaAssembly, config.versions(), null, 0, supplier.sharedEnvironmentProvider);
```

Comment:
I wonder if you can get rid of this.

Reply:
Later in the `deletePersistentClaims` method it's used to get the selector labels.

Reply:
Just saw your comment later :-)

Reply:
I got rid of this one.